ev3dev/auto.py

```diff
@@ -1,20 +1,40 @@
-import platform
 
-# -----------------------------------------------------------------------------
-# Guess platform we are running on
+"""
+Use platform.machine() to determine the platform type, cache the
+results in /tmp/current_platform so that we do not have to import
+platform and run platform.machine() each time someone imports ev3dev.auto
+"""
+import os
 
-def current_platform():
-    machine = platform.machine()
-    if machine == 'armv5tejl':
-        return 'ev3'
-    elif machine == 'armv6l':
-        return 'brickpi'
-    else:
-        return 'unsupported'
+filename = '/tmp/current_platform'
+current_platform = None
 
-# -----------------------------------------------------------------------------
+if os.path.exists(filename):
+    with open(filename, 'r') as fh:
+        current_platform = fh.read().strip()
 
-if current_platform() == 'brickpi':
+if not current_platform:
+    import platform
+
+    def get_current_platform():
+        """
+        Guess platform we are running on
+        """
+        machine = platform.machine()
+
+        if machine == 'armv5tejl':
+            return 'ev3'
+        elif machine == 'armv6l':
+            return 'brickpi'
+        else:
+            return 'unsupported'
+
+    current_platform = get_current_platform()
+
+    with open(filename, 'w') as fh:
+        fh.write(current_platform + '\n')
+
+if current_platform == 'brickpi':
```

Comment:
These changes allow us to avoid import platform and running platform.machine() every time the user runs a program that imports ev3dev.auto.  This saves us about 1s.